### PR TITLE
accept block when producing

### PR DESCRIPTION
**Pull request by ghost**
## Change Description

### CODE REVIEW

File: libraries/chain/controller.cpp [hunk 1]
```diff
@@ -2144,9 +2134,36 @@ struct controller_impl {
...
+         if (self.is_trusted_producer(b->producer)) {
+            trusted_producer_light_validation = true;
+         };
+
+         emit( self.accepted_block_header, bsp );
+
+         if( read_mode != db_read_mode::IRREVERSIBLE ){
             //main node receive backup block
             fc_dlog(_backup_block_trace_log,"[BACKUP_TRACE] main producer node receive backup block....");
             bool broadcast = false;
```
Comment (metamaster2021) [inline libraries/chain/controller.cpp]: rename broacast to "is_accepted_block"

File: libraries/chain/controller.cpp [hunk 2]
```diff
@@ -3398,8 +3433,8 @@ bool controller::is_producing_block()const {
    return (my->pending->_block_status == block_status::incomplete);
 }
 
-bool controller::is_backup_produce()const{
-   return is_backup_mode;
+bool controller::is_backup_producing()const{
```
Comment (metamaster2021) [inline libraries/chain/controller.cpp]: should rename 'is_backup_producing' to 'pending_block_is_backup'

File: plugins/producer_plugin/producer_plugin.cpp [hunk 3]
```diff
@@ -341,10 +333,49 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
...
+               if( chain.get_read_mode() != db_read_mode::IRREVERSIBLE && hbs->id != id && hbs->block != nullptr ) { // not applied to head
+                  ilog("Block not applied to head ${id}... #${n} @ ${t} signed by ${p} [trxs: ${count}, dpos: ${dpos}, conf: ${confs}, latency: ${latency} ms]",
+                     ("p",hbs->block->producer)("id",hbs->id.str().substr(8,16))("n",hbs->block_num)("t",hbs->block->timestamp)
+                     ("count",hbs->block->transactions.size())("dpos", hbs->dpos_irreversible_blocknum)
+                     ("confs", hbs->block->confirmed)("latency", (fc::time_point::now() - hbs->block->timestamp).count()/1000 ) );
+               }
+            }
+            return true;
+         }
+         if ( _pending_block_mode == pending_block_mode::producing && chain.is_producing_block() && !chain.is_backup_producing() && chain.pending_block_producer() != block->producer ) {
```
Comment (metamaster2021) [inline plugins/producer_plugin/producer_plugin.cpp]: remove condition 'chain.pending_block_producer() != block->producer'

Comment (ghost) [reply plugins/producer_plugin/producer_plugin.cpp]: can not remove this condition , if remove this condition the main node cann't sync from backup node again

Comment (ghost) [reply plugins/producer_plugin/producer_plugin.cpp]: 1st: set enable-stale-production = false by mannally.
wait sync
2nd: set enable-stale-production = true by mannally.

File: libraries/chain/controller.cpp (hunk 1 above)
Comment (ghost) [reply libraries/chain/controller.cpp]: modified

**Review by metamaster2021 — APPROVED** [review]

**Pull request merged**